### ISSUE TITLE
Bug: DescriptorSet update of multiple bindings with one update is incorrect

### ISSUE THREAD

**Issue by quehuo01** [KhronosGroup/Vulkan-ValidationLayers#2909]
**Describe the Issue**
The bahavior of the layers are incorrect when updating a DescriptorSet with multiple bindings.
Our example is (Binding list):
1. type: UNIFORM_STORAGE_BUFFER stage:compute descriptorCount: 16
2. type: SOMETHING_ELSE stage:compute descriptorCount: 0
3. type: UNIFORM_STORAGE_BUFFER stage:compute descriptorCount: 16

A descriptor set update on the binding 1 with a descriptor count > 16 (&& < 32) will fail

However the spec says that binding 2 in this case should be skipped https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkWriteDescriptorSet-descriptorCount-00317

**Valid Usage ID**
```
'Validation' reported [ error ]:
- Message: 'Validation Error: [ VUID-VkWriteDescriptorSet-descriptorCount-00317 ] Object 0: handle = 0x65de03000001f4fb, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xef042099 | vkUpdateDescriptorSets() pDescriptorWrites[0] failed write update validation for VkDescriptorSet 0x65de03000001f4fb[] with error: Attempting write update to VkDescriptorSet 0x65de03000001f4fb[] allocated with VkDescriptorSetLayout 0x8ef6ba000001f4fa[] binding index #1 (16 from dstBinding offset) with a different stageFlag and/or descriptorType from previous bindings. All bindings must have consecutive stageFlag and/or descriptorType across a VkWriteDescriptorSet. The Vulkan spec states: All consecutive bindings updated via a single VkWriteDescriptorSet structure, except those with a descriptorCount of zero, must have identical descriptorType and stageFlags (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkWriteDescriptorSet-descriptorCount-00317)'
- Flags: 8 objectType: 23 object: 7340307741195498747 location: 18446744073424609433 messageCode: 0
```

**Environment:**
 - OS: Linux
 - GPU: N/A
 - SDK or header version if building from repo:
 - Options enabled (synchronization, best practices, etc.):

**Additional context**

**Comment by ziga-lunarg** [comment]
@quehuo01 This has been closed by #3267, correct?

**Comment by quehuo01** [comment]
Yes it is closed by #3267
